Snakemake/workflow/rules/prepare_files_for_tsinfer.smk

```diff
@@ -64,23 +65,41 @@ rule extract_vcf_pos:
         rm tmp
         """
 
+# I changed here so the rule now takes as input
+#   (1) positions in the vcf
+#   (2) ancestral allele
+#   (3) major allele
 rule match_ancestral_vcf:
     input:
         vcfPos=rules.extract_vcf_pos.output, # This has more lines,
-        ancestralMajor=rules.combine_major_ancestral.output
+        ancestral=config['ancestralAllele'],
+        major=rules.get_major.output,
+#        ancestralMajor=rules.combine_major_ancestral.output
         # The ancestral file has to have chr_pos and AA, split with a tab
     output: temp('AncestralVcfMatch{chromosome}.txt')
         # THis files needs to contain all the variants from the vcf (blank space)
     shell:
+    # take only the ancestrals referent to the current chr
+    # read the position file (1) line by line;
+    # test if the position existis in the ancestral file (2),
+    # if TRUE print the line from file (2),
+    # else print the line from file (3)
         """
+        grep "{wildcards.chromosome}_" {input.ancestral} > aa_tmp
+
         for line in $(cat {input.vcfPos});
         do
-          grep $line {input.ancestralMajor} || echo "";
-        done > tmp
-        awk -F"," '{{print $1"\t"$2}}' tmp > {output}
+            if grep -wq "${line}" aa_tmp; then
```

Comment:
Ok - I don't really understand, are we changing it or we leave it like this? Honestly, I don't know how long it takes to grep something from a huge file - I'm just worried we might be greping for nothing because if it finds something, then we need to grep again.

Reply:
we can change

Reply:
@janaobsteter I updated this, so now if grep returns true, it directly adds the line to the output file `then continue` (`continue` is always a great trick)